app/workers/publishing_api_document_republishing_worker.rb

```diff
@@ -38,9 +38,7 @@ def perform(document_id, bulk_publishing = false)
       document.lock!
 
       if the_document_has_been_unpublished?
-        send_draft_and_unpublish
-      elsif the_document_has_been_withdrawn?
-        send_published_and_withdraw
+        send_published_and_unpublish
```

Comment:
Yeah I wasn't 100% sure on this. I think it does need to be updated, but I might have made the wrong change, thinking about it. I was working based on the fact that there was previously code to handle unpublished documents,  which rerendered and sent the draft that was created during unpublishing and resent the "unpublishing" data to Publishing API. There isn't a draft to send any longer, so this doesn't make sense and I think should be changed.

I based the new behaviour on what happens with withdrawn documents. For withdrawn documents, it resends the document, which I believe republishes it to Publishing API, and then resends the unpublishing to ensure it's immediately unpublished again. However, because unpublished documents are not rendered, unlike withdrawn documents, that isn't really necessary. I think for an unpublished document the worker should really perform a no-op. Would you agree?

Reply:
Reverted these changes

Reply:
We ended up sending the unpublishing request again so that the unpublishing explanation can be updated if necessary, and the sending the draft if there is one.